cli/cmd/generate_aws.go

```diff
@@ -0,0 +1,375 @@
+		Opts:     []survey.AskOpt{survey.WithValidator(validPathExists)},
+		Required: true,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func askAdvancedOptions(config *aws.GenerateAwsTfConfigurationArgs, extraState *AwsGenerateCommandExtraState) error {
+	answer := ""
+
+	// Prompt for options
+	for answer != AdvancedOptDone {
+		// Construction of this slice is a bit strange at first look, but the reason for that is because we have to do string
+		// validation to know which option was selected due to how survey works; and doing it by index (also supported) is
+		// difficult when the options are dynamic (which they are)
+		//
+		// Only ask about more accounts if consolidated cloudtrail is setup (matching scenarios doc)
+		options := []string{AdvancedOptCloudTrail, AdvancedOptIamRole}
+		if config.ConsolidatedCloudtrail {
+			options = append(options, AdvancedOptAwsAccounts)
+		}
+		options = append(options, AdvancedOptLocation, AdvancedOptDone)
+		if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+			Prompt: &survey.Select{
+				Message: "Which options would you like to enable?",
+				Options: options,
+			},
+			Response: &answer,
+		}); err != nil {
+			return err
+		}
+
+		// Based on response, prompt for actions
+		switch answer {
+		case AdvancedOptCloudTrail:
+			if err := promptAwsCtQuestions(config, extraState); err != nil {
+				return err
+			}
+		case AdvancedOptIamRole:
+			if err := promptAwsExistingIamQuestions(config); err != nil {
+				return err
+			}
+		case AdvancedOptAwsAccounts:
+			if err := promptAwsAdditionalAccountQuestions(config); err != nil {
+				return err
+			}
+		case AdvancedOptLocation:
+			if err := promptCustomizeOutputLocation(config, extraState); err != nil {
+				return err
+			}
+		}
+
+		// Re-prompt if not done
+		innerAskAgain := true
+		if answer == AdvancedOptDone {
+			innerAskAgain = false
+		}
+
+		if err := SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+			Checks:   []*bool{&innerAskAgain},
+			Prompt:   &survey.Confirm{Message: QuestionAnotherAdvancedOpt, Default: false},
+			Response: &innerAskAgain,
+		}); err != nil {
+			return err
+		}
+
+		if !innerAskAgain {
+			answer = AdvancedOptDone
+		}
+	}
+
+	return nil
+}
+
+func configOrCloudtrailEnabled(config *aws.GenerateAwsTfConfigurationArgs) *bool {
+	cloudtrailOrConfigEnabled := config.Cloudtrail || config.Config
+	return &cloudtrailOrConfigEnabled
+}
+
+// entry point for launching a survey to build out the required generation parameters
+func promptAwsGenerate(
+	config *aws.GenerateAwsTfConfigurationArgs,
+	existingIam *aws.ExistingIamRoleDetails,
+	extraState *AwsGenerateCommandExtraState,
+) error {
+	// Cache for later use if generation is abandon and in interactive mode
+	if cli.InteractiveMode() {
+		defer cli.WriteAssetToCache("iac-aws-generate-params", time.Now().Add(time.Hour*1), config)
+		defer cli.WriteAssetToCache("extra-state", time.Now().Add(time.Hour*1), extraState)
```

Comment:
I don't like the name `extra-state` but I understand what it is, could we at least add a prefix in case other providers also have an "extra state"?
```suggestion
		defer cli.WriteAssetToCache("iac-aws-extra-state", time.Now().Add(time.Hour*1), extraState)
```

Reply:
Also, could we move these names to the top and have them as constants?

Reply:
👍  yep & yep; will update